core/transfer/archive/importer.go

```diff
@@ -64,7 +65,14 @@ func (iis *ImageImportStream) Import(ctx context.Context, store content.Store) (
 	if iis.forceCompress {
 		opts = append(opts, archive.WithImportCompression())
 	}
-	return archive.ImportIndex(ctx, store, iis.stream, opts...)
+
+	r, err := compression.DecompressStream(iis.stream)
```

Comment:
Should we be making use of the media type here? Possibly only running through this logic if the media type indicates compression or is empty.

Reply:
We can check for empty media type, but here the input is a compressed OCI tarball (with possibly uncompressed layers). Typically the media type here is `ocispec.MediaTypeImageIndex` and there isn't a compressed version afaik.